### DISCUSSION TITLE
netbox.servis not working after trying to install plugin

### DISCUSSION THREAD

**Discussion by Fullag1** [netbox-community/netbox#17540] [Help Wanted!] [answered]
i tried to install netbox-proxbox. after i did ` $ systemctl restart netbox` it stop working. when I go to the page it gives me **502 Bad Gateway**
i did
`$ systemctl status netbox.service`
i got this result
`Loaded: loaded (/etc/systemd/system/netbox.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Thu 2024-09-19 06:37:00 UTC; 6s ago
       Docs: https://docs.netbox.dev/
    Process: 117352 ExecStart=/opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi (code=exited, status=1/FAILURE)
   Main PID: 117352 (code=exited, status=1/FAILURE)
        CPU: 2.467s`
I am not a very experienced Linux user, so I don't know how serious the problem is.

**Comment by a084ed22** [comment]
Can you try to check what the logs say via `journalctl -u netbox.service` ?

**Reply by Fullag1** [reply]
sure!


Sep 19 06:58:36 netbox systemd[1]: netbox.service: Main process exited, code=exited, status=1/FAILURE
Sep 19 06:58:36 netbox gunicorn[123429]: [2024-09-19 06:58:36 +0000] [123429] [ERROR] Exception in worker process
Sep 19 06:58:36 netbox gunicorn[123429]: Traceback (most recent call last):
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/netbox/netbox/settings.py", line 797, in <module>
Sep 19 06:58:36 netbox gunicorn[123429]:     plugin = importlib.import_module(plugin_name)
Sep 19 06:58:36 netbox gunicorn[123429]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
Sep 19 06:58:36 netbox gunicorn[123429]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 19 06:58:36 netbox gunicorn[123429]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
Sep 19 06:58:36 netbox gunicorn[123429]: ModuleNotFoundError: No module named 'netbox_proxmox_sync'
Sep 19 06:58:36 netbox gunicorn[123429]: During handling of the above exception, another exception occurred:
Sep 19 06:58:36 netbox gunicorn[123429]: Traceback (most recent call last):
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/arbiter.py", line 608, in spawn_worker
Sep 19 06:58:36 netbox gunicorn[123429]:     worker.init_process()
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/workers/gthread.py", line 94, in init_process
Sep 19 06:58:36 netbox gunicorn[123429]:     super().init_process()
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/workers/base.py", line 135, in init_process
Sep 19 06:58:36 netbox gunicorn[123429]:     self.load_wsgi()
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/workers/base.py", line 147, in load_wsgi
Sep 19 06:58:36 netbox gunicorn[123429]:     self.wsgi = self.app.wsgi()
Sep 19 06:58:36 netbox gunicorn[123429]:                 ^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/app/base.py", line 66, in wsgi
Sep 19 06:58:36 netbox gunicorn[123429]:     self.callable = self.load()
Sep 19 06:58:36 netbox gunicorn[123429]:                     ^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 57, in load
Sep 19 06:58:36 netbox gunicorn[123429]:     return self.load_wsgiapp()
Sep 19 06:58:36 netbox gunicorn[123429]:            ^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 47, in load_wsgiapp
Sep 19 06:58:36 netbox gunicorn[123429]:     return util.import_app(self.app_uri)
Sep 19 06:58:36 netbox gunicorn[123429]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/gunicorn/util.py", line 370, in import_app
Sep 19 06:58:36 netbox gunicorn[123429]:     mod = importlib.import_module(module)
Sep 19 06:58:36 netbox gunicorn[123429]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
Sep 19 06:58:36 netbox gunicorn[123429]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 19 06:58:36 netbox gunicorn[123429]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap_external>", line 995, in exec_module
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/netbox/netbox/wsgi.py", line 7, in <module>
Sep 19 06:58:36 netbox gunicorn[123429]:     application = get_wsgi_application()
Sep 19 06:58:36 netbox gunicorn[123429]:                   ^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
Sep 19 06:58:36 netbox gunicorn[123429]:     django.setup(set_prefix=False)
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/django/__init__.py", line 19, in setup
Sep 19 06:58:36 netbox gunicorn[123429]:     configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
Sep 19 06:58:36 netbox gunicorn[123429]:                       ^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/django/conf/__init__.py", line 89, in __getattr__
Sep 19 06:58:36 netbox gunicorn[123429]:     self._setup(name)
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/django/conf/__init__.py", line 76, in _setup
Sep 19 06:58:36 netbox gunicorn[123429]:     self._wrapped = Settings(settings_module)
Sep 19 06:58:36 netbox gunicorn[123429]:                     ^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/venv/lib/python3.12/site-packages/django/conf/__init__.py", line 190, in __init__
Sep 19 06:58:36 netbox gunicorn[123429]:     mod = importlib.import_module(self.SETTINGS_MODULE)
Sep 19 06:58:36 netbox gunicorn[123429]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
Sep 19 06:58:36 netbox gunicorn[123429]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 19 06:58:36 netbox gunicorn[123429]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap_external>", line 995, in exec_module
Sep 19 06:58:36 netbox gunicorn[123429]:   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
Sep 19 06:58:36 netbox gunicorn[123429]:   File "/opt/netbox/netbox/netbox/settings.py", line 800, in <module>
Sep 19 06:58:36 netbox gunicorn[123429]:     raise ImproperlyConfigured(
Sep 19 06:58:36 netbox gunicorn[123429]: django.core.exceptions.ImproperlyConfigured: Unable to import plugin netbox_proxmox_sync: Module not found. Check that the plugin module has been installed within the correct Python environment.
Sep 19 06:58:36 netbox gunicorn[123429]: [2024-09-19 06:58:36 +0000] [123429] [INFO] Worker exiting (pid: 123429)
Sep 19 06:58:36 netbox systemd[1]: netbox.service: Failed with result 'exit-code'.
Sep 19 06:58:36 netbox systemd[1]: netbox.service: Consumed 2.459s CPU time.

**Reply by a084ed22** [reply, marked as answer]
You seem to have the netbox-proxmox-sync module either referenced but not installed or partially installed. You should either disable the plugin and remove references from configuration.py, or revise your setup.

**Reply by Fullag1** [reply]
Ok, thanks! I'll let you know if anything changes.

**Reply by Fullag1** [reply]
ok, it works again, I understood that this config.py does not play such a big role, but still I disabled the settings that I made earlier, went to "root" and re-enabled the service. Thanks you mate!
